### ISSUE TITLE
Scope being overwritten

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/crbuilder#6]
ID: 07bf0e3a-e2db-4887-b843-ef8b48b58934
Rule Id: CG0143

Prod Scope:
```yaml
Scope:
  Classes:
    Include:
      - EVENTS
  Domains:
    Include:
      - DS
```

Generated Scope:
```yaml
Scope:
  Classes:
    Include:
      - SPECIAL PURPOSE
  Domains:
    Include:
      - DM
```

Again, no properties should be overwritten. The merge should only do additions, not replaces. For this case, where the scopes don't match what is expected, this should appear in the merge log file for manual investigation.

**Comment by htu** [comment]
This is what is in the rule definition: 

CG0143 | 3.2 | 1 | SPC | DM
-- | -- | -- | -- | --
CG0143 | 3.3 | 1 | SPC | DM
CG0143 | 3.4 | 1 | SPC | DM

Does this mean rule author can change the rule definition?

**Comment by htu** [comment]
I have changed the code so that it will not overwritten if the scope is already created in the rule.

**Comment by gerrycampion** [comment]
> This is what is in the rule definition:
> 
> CG0143	3.2	1	SPC	DM
> CG0143	3.3	1	SPC	DM
> CG0143	3.4	1	SPC	DM
> Does this mean rule author can change the rule definition?

Yes, the author can write the rule differently from what is in the spec verbatim and we want to see these cases in the logs.

In this example, the author correctly changed the scope because the rule condition is based on DS and references DM instead of the other way around.

**Comment by htu** [comment]
This has been fixed in v0.4.1.
